device/cuda/src/finding/finding_algorithm.cu

```diff
@@ -65,12 +65,14 @@ __global__ void make_barcode_sequence(
 /// CUDA kernel for running @c traccc::device::apply_interaction
 template <typename detector_t>
 __global__ void apply_interaction(
-    typename detector_t::view_type det_data, const int n_params,
+    typename detector_t::view_type det_data, const finding_config& cfg,
```

Comment:
Can you pass the config as a reference to a kernel?

Reply:
In fact that's a very good catch!

```suggestion
    typename detector_t::view_type det_data, const finding_config cfg,
```

Reply:
you mean pass the config as a value?

Reply:
Yes, you need to get the data in the ```finding_config``` to the device somehow, right?

Reply:
:confused: I gave a code change suggestion.

You cannot pass a const reference to a kernel. That's exactly the same as passing a const pointer. Which is pointing at an object that sits on the stack. **On the host.** You need to pass config objects like this by value to the kernel.

From the kernel then you can pass objects forward using references. It's only on the host<->device boundary that you need to be careful.

Reply:
Bro I said that I am ignorant of this.. You can't just say "I gave a code to you".. 

Thanks for the explanation though

Reply:
Ah, sorry, I thought you just overlooked this and that it would be clear if I highlight the line, my bad

Reply:
? what are you apologizing for lol

Reply:
It is fixed now